Re-enable test_copy_transpose_math_view, neg_view/dce fix

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #110770
* __->__ #110651



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

- neg view can just be lowered to neg() post functionalization
- we were treating all fallback kernels as not having side effects. we shouldn't dce mutating fallback kernels - either mutations induced by the reinplacing pass or clone_ with unsupported arguments (complex)

eellison committed:
Re-enable test_copy_transpose_math_view, neg_view/dce fix

[ghstack-poisoned]

eellison committed:
Update on "Re-enable test_copy_transpose_math_view, neg_view/dce fix"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

- neg view can just be lowered to neg() post functionalization
- we were treating all fallback kernels as not having side effects. we shouldn't dce mutating fallback kernels - either mutations induced by the reinplacing pass or clone_ with unsupported arguments (complex)

[ghstack-poisoned]

Chillee commented:
> either mutations induced by the reinplacing pass 

Wait sorry, when can this result in inplace fallback kernels?

Chillee reviewed (approved):
Oh I see. 🤔

eellison committed:
Update on "Re-enable test_copy_transpose_math_view, neg_view/dce fix"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

- neg view can just be lowered to neg() post functionalization
- we were treating all fallback kernels as not having side effects. we shouldn't dce mutating fallback kernels - either mutations induced by the reinplacing pass or clone_ with unsupported arguments (complex)

[ghstack-poisoned]

eellison committed:
Update on "Re-enable test_copy_transpose_math_view, neg_view/dce fix"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

- neg view can just be lowered to neg() post functionalization
- we were treating all fallback kernels as not having side effects. we shouldn't dce mutating fallback kernels - either mutations induced by the reinplacing pass or clone_ with unsupported arguments (complex)

[ghstack-poisoned]

eellison committed:
Update on "Re-enable test_copy_transpose_math_view, neg_view/dce fix"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

- neg view can just be lowered to neg() post functionalization
- we were treating all fallback kernels as not having side effects. we shouldn't dce mutating fallback kernels - either mutations induced by the reinplacing pass or clone_ with unsupported arguments (complex)

[ghstack-poisoned]

eellison committed:
Update on "Re-enable test_copy_transpose_math_view, neg_view/dce fix"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

- neg view can just be lowered to neg() post functionalization
- we were treating all fallback kernels as not having side effects. we shouldn't dce mutating fallback kernels - either mutations induced by the reinplacing pass or clone_ with unsupported arguments (complex)

[ghstack-poisoned]

eellison committed:
Update on "Re-enable test_copy_transpose_math_view, neg_view/dce fix"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

- neg view can just be lowered to neg() post functionalization
- we were treating all fallback kernels as not having side effects. we shouldn't dce mutating fallback kernels - either mutations induced by the reinplacing pass or clone_ with unsupported arguments (complex)

[ghstack-poisoned]

jansel commented on torch/_inductor/debug.py:
```diff
@@ -451,6 +451,7 @@ def ir_post_fusion(self, nodes: SchedulerNodeList):
 
     def _write_ir(self, filename: str, nodes: SchedulerNodeList):
         with self.fopen(filename) as fd:
+            print(f"Writing debug ir to {fd.name}")
```
remove or log.info?

malfet commented on torch/_inductor/debug.py:
```diff
@@ -451,6 +451,7 @@ def ir_post_fusion(self, nodes: SchedulerNodeList):
 
     def _write_ir(self, filename: str, nodes: SchedulerNodeList):
         with self.fopen(filename) as fd:
+            print(f"Writing debug ir to {fd.name}")
```
Not needed I guess?
```suggestion
```

eellison replied:
This is only generated if debug logs are already set. Currently we were writing them out but not specifying where they were written so you couldnt find them. I can switch this to using logging though.

malfet replied:
Still I think `print` is worse than using a structured logging (@ezyang knows best which mechanism is preferred right now)

malfet commented on torch/_inductor/ir.py:
```diff
@@ -3846,6 +3847,19 @@ def find_device(tensor_args, example_output):
+        # TODO - some fallbacks are still OpOverloadPackets
+        if not isinstance(self.op_overload, torch._ops.OpOverload):
+            return False
+
```
Nit (or add newline between similar calls on line 3853
```suggestion
```

eellison committed:
Update on "Re-enable test_copy_transpose_math_view, neg_view/dce fix"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

- neg view can just be lowered to neg() post functionalization
- we were treating all fallback kernels as not having side effects. we shouldn't dce mutating fallback kernels - either mutations induced by the reinplacing pass or clone_ with unsupported arguments (complex)

[ghstack-poisoned]

eellison commented:
@pytorchbot merge -f "Flakey distributed failure"
